app/controllers/likes_controller.rb

```diff
@@ -0,0 +1,19 @@
+class LikesController < ApplicationController
+  before_action :set_post
+
+  def create
+    @like = Like.create(user_id: current_user.id, post_id: @post.id)
+    redirect_back(fallback_location: post_path(params[:post_id]))
+  end
+
+  def destroy
+    @like = Like.find_by(user_id: current_user.id, post_id: @post.id)
```

Comment:
destroyメソッドでは、インスタンス変数を定義する必要はないはずです（ビュー側に渡さないので）。
例えば以下の書き方などすれば記述量を減らせます！
```suggestion
    Like.find_by(id: params[:id]).destroy
```

Reply:
`Like.find_by(user_id: current_user.id, post_id: @post.id).destroy`に変更しました

Like.find_by(id: params[:id]).destroyの場合は、post_controller上で@likeを定義する必要があるので、そのままにしています。

Reply:
了解ですー！